src/field/scrcmd_mystery_gift.c

```diff
@@ -0,0 +1,684 @@
+        pFunc->messageSuccess(&gmmState, pMsgBank, pMsgNum);
+        break;
+    }
+    case SCR_MG_MESSAGE_FAILED: {
+        struct GetMysteryGiftGmmState gmmState;
+        const struct ScriptMysteryGiftFuncs* pFunc = &sScriptMysteryGiftActionTable[FieldSystem_GetTagOfNextMG(ctx->fieldSystem) - 1];
+        MessageFormat** pMsgFormat = FieldSysGetAttrAddr(ctx->fieldSystem, SCRIPTENV_MESSAGE_FORMAT);
+        u16* pMsgBank = ScriptGetVarPointer(ctx);
+        u16* pMsgNum = ScriptGetVarPointer(ctx);
+        FieldSystem_InitGetMysteryGiftGmmState(&gmmState, ctx->fieldSystem, *pMsgFormat, FieldSystem_GetDataOfNextMG(ctx->fieldSystem));
+        pFunc->messageFailure(&gmmState, pMsgBank, pMsgNum);
+        break;
+    }
+    }
+    return FALSE;
+}
+
+BOOL MGCheck_PartySpace(FieldSystem* fieldSys, MysteryGiftData* unused) {
+    (void)unused;
+    return Party_GetCount(SaveArray_Party_Get(fieldSys->saveData)) < PARTY_SIZE;
+}
+
+void MGGive_ManaphyEgg(FieldSystem* fieldSys, MysteryGiftData* unused) {
+    (void)unused;
+    GiveEgg(HEAP_ID_32, fieldSys->saveData, SPECIES_MANAPHY, MAPSEC_TWINLEAF_TOWN, MAPSECTYPE_EXTERNAL, MAPLOC(METLOC_POKEMON_RANGER));
+}
+
+void MGMessageSuccess_ManaphyEgg(struct GetMysteryGiftGmmState* gmmState, u16* pMsgBank, u16* pMsgNum) {
+    *pMsgBank = NARC_msg_msg_0209_bin;
+    *pMsgNum = msg_0209_00013;
+    BufferPlayersName(gmmState->msgFormat, 0, Save_PlayerData_GetProfileAddr(gmmState->fieldSys->saveData));
+}
+
+void MGGive_Mon(FieldSystem* fieldSys, MysteryGiftData* unused) {
+    (void)unused;
+
+    MysteryGiftPokemonTag* mgData = &FieldSystem_GetDataOfNextMG(fieldSys)->pokemon;
+    PlayerProfile* profile = Save_PlayerData_GetProfileAddr(fieldSys->saveData);
+    SaveVarsFlags* vars_flags = Save_VarsFlags_Get(fieldSys->saveData);
+    Pokemon* tmpPokemon = NULL;
+    Pokemon* pokemon = &mgData->mon;
+    u8* srcRibbons = mgData->ribbons;
+    int eggMetLocation = GetMonData(pokemon, MON_DATA_EGG_MET_LOCATION, NULL);
+    int personality = GetMonData(pokemon, MON_DATA_PERSONALITY, NULL);
+    int otid = GetMonData(pokemon, MON_DATA_OTID, NULL);
+    int rand = PRandom(OS_GetTick());
+
+    if (personality != 0) {
+        if (personality == 1) {
+            while (CalcShininessByOtIdAndPersonality(otid, rand)) {
+                rand = PRandom(rand);
+            }
+        } else {
+            rand = personality;
+        }
+    }
+    SetMonPersonality(pokemon, rand);
+
+    int gender = GetMonGender(pokemon);
+    SetMonData(pokemon, MON_DATA_GENDER, &gender);
+
+    gender = GetMonData(pokemon, MON_DATA_HP_IV, NULL)
+        + GetMonData(pokemon, MON_DATA_ATK_IV, NULL)
+        + GetMonData(pokemon, MON_DATA_DEF_IV, NULL)
+        + GetMonData(pokemon, MON_DATA_SPEED_IV, NULL)
+        + GetMonData(pokemon, MON_DATA_SPATK_IV, NULL)
+        + GetMonData(pokemon, MON_DATA_SPDEF_IV, NULL);
+    if (gender == 0) {
+        u16 ivRand = LCRandom();
+        rand = ivRand & 0x1Fu;
```

Comment:
any significance to these values? or are they just arbritrary

Reply:
not sure what you're asking. did this comment end up on the wrong line?